src/libs/masq.js

```diff
@@ -84,25 +84,18 @@ export default class MasqStore {
 
   async getAllPois() {
     await this.checkInit()
-    const list = await this.masq.list()
 
-    const filteredKeys = Object.keys(list).reduce((filtered, k) => {
-      if (Poi.isPoiCompliantKey(k)) {
-        filtered.push(k)
-      }
-      return filtered
-    }, [])
-
-    if (filteredKeys.length === 0) {
-      return []
-    }
+    try {
+      const list = await this.masq.list()
 
-    const valuePromises = filteredKeys.map(k => {
-      return this.masq.get(k)
-    })
+      const filteredValues = Object.entries(list).reduce((filtered, kv) => {
```

Comment:
I feel that its more readable to replace the generic reduce by a more specific [map] ◯ [filter]
you can avoid duplicated variable

Reply:
The reduce should be more efficient than applying filter and map, but I can change it if you think it would be more readable ?
It would be something like
`Object.entries(list).filter(kv => Poi.isPoiCompliantKey(kv[0])).map(kv => kv[1])`

Reply:
Yep more readable maybe add some new line :)